pkg/util/lifted/lua/loadlib_safe.go

```diff
@@ -0,0 +1,86 @@
+/*
+The MIT License (MIT)
+
+Copyright (c) 2015 Yusuke Inuzuka
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+*/
+
+package lua
+
+// This code is directly lifted from the github.com/yuin/gopher-lua codebase in order to use the Lua interpreter as safely as possible.
+// We do not allow users to load modules using the path patterns defined in package.path (for Lua files) and package.cpath (for C libraries) when interpreting resources with lua scripts, which may potentially grant providers access to modules loaded outside the sandbox.
```

Comment:
We have set UseOpenLibs to true by default, which will not invoke the following packages in the gopher-lua library:

https://github.com/yuin/gopher-lua/blob/2348fd042596f47bf9a1018a01f365ca3bf19134/linit.go#L31-L42

But we have displayed the opening of these packages:

https://github.com/karmada-io/karmada/blob/caf34c6c7ff0f07e44c6d13a31195fe6e5f418cd/pkg/resourceinterpreter/customized/declarative/luavm/lua.go#L184-L190

Some functions in the package and base are unsafe, please refer to the [documentation](https://lua-users.org/wiki/SandBoxes). I am currently unsure if the vm created by gopher-lua is completely safe and able to prevent attacks, it is currently under analysis. 

Have you analyzed this issue before?

Reply:
ohh,  Thank you for the reminder.

> Have you analyzed this issue before?


No,  I haven't analyzed that issue either.